### DISCUSSION TITLE
[@types/supertest] Error transpiling with TypeScript 4.6.3

### DISCUSSION THREAD

**Discussion by zebrapurring** [DefinitelyTyped/DefinitelyTyped#69863] [Issues with a @types package] [answered]
* Node version: v22.2.0
* TypeScript version: 4.6.3
* Supertest version: 7.0.0
* `@types/supertest` version: 6.0.2

Generates the following error when running `tsc`:

```
node_modules/@types/supertest/types.d.ts:9:7 - error TS2315: Type 'RequestListener' is not generic.

9     | RequestListener<typeof IncomingMessage, typeof ServerResponse>
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in node_modules/@types/supertest/types.d.ts:9
```

[a comment by an automated account is not shown]

**Comment by DavidTanner** [comment, marked as answer]
@zebrapurring Can you check your version of `@types/node`?  Either it is old, or the new http has a different `RequestListener` type.

**Reply by zebrapurring** [reply]
You're right! I was still on `@types/node` v17, upgrading to v20 solved the issue 🙂

**Comment by DavidTanner** [comment]
From the source it looks like it has been been parameterized for the last 2 years, since node 16.
